src/scenic/core/pruning.py

```diff
@@ -150,21 +223,30 @@ def pruneRelativeHeading(scenario, verbosity):
     of F which satisfy the relative heading requirements w.r.t. some cell of F' which
     is within the distance bound.
     """
+    # TODO Add test for empty pruned polygon (Might cause crash?)
     # Check which objects are (approximately) aligned to polygonal vector fields
     fields = {}
     for obj in scenario.objects:
         field, offsetL, offsetR = matchPolygonalField(obj.heading, obj.position)
         if field is not None:
             fields[obj] = (field, offsetL, offsetR)
+
     # Check for relative heading relations among such objects
     for obj, (field, offsetL, offsetR) in fields.items():
         position = currentPropValue(obj, 'position')
-        base = matchInRegion(position)
-        if base is None:        # obj must be positioned uniformly in a Region
+        base, _, _, offset = matchInRegion(position)
+
+        if base is None or needsSampling(base):        # obj must be positioned uniformly in a Region
+            continue
+
+        if offset is not None:
             continue
+
         basePoly = regions.toPolygon(base)
+
```

Comment:
Let's not add gratuitous blank lines like this -- I don't think it makes the code any easier to read. Blank lines are fine for separating parts of the code that do quite different things, but separating a definition from where it is used doesn't make sense to me.